03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,421 @@
+
+In this analysis, we will be using the R package [`DESeq2`](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html) [@DESeq2] for normalization and the R package `pheatmap` [@pheatmap] for clustering and creating a heatmap.
+
+```{r}
+if (!("pheatmap" %in% installed.packages())) {
+  # Install pheatmap
+  install.packages("pheatmap", update = FALSE)
+}
+
+if (!("DESeq2" %in% installed.packages())) {
+  # Install DESeq2
+  BiocManager::install("DESeq2", update = FALSE)
+}
+```
+
+Attach the `pheatmap` and `DESeq2` libraries:
+
+```{r}
+# Attach the `pheatmap` library
+library(pheatmap)
+
+# Attach the `DESeq2` library
+library(DESeq2)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Set the seed so our results are reproducible:
+set.seed(12345)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata [tab separated values ("tsv")](https://en.wikipedia.org/wiki/Tab-separated_values) file and a data tsv file.
+This chunk of code will read in both tsv files and add them as data.frames to your environment.
+
+```{r}
+# Read in metadata tsv file
+metadata <- readr::read_tsv(file.path(
+  "data", # Replace with path to your metadata file
+  "metadata_SRP070849.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data tsv file
+df <- readr::read_tsv(file.path(
+  "data", # Replace with path to your data file
+  "SRP070849.tsv" # Replace with the name of your data file
+)) %>%
+  # Here we are going to store the gene IDs as rownames so that we can have a numeric matrix to perform calculations on later
+  tibble::column_to_rownames("Gene")
+```
+
+Let's take a look at the metadata object that we read into the R environment.
+
+```{r}
+head(metadata)
+```
+
+Now let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>% dplyr::select(metadata$refinebio_accession_code)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+# How are our samples clustering together?
+
+Now we are going to use a combination of functions from the `DESeq2` and `pheatmap` packages to look at how are samples and genes are clustering.
+
+## Create a DESeqDataset
+
+We will be using the `DESeq2` package for normalization, which requires us to format our data into a `DESeqDataSet` object.
+We turn the data.frame (or matrix) into a [`DESeqDataSet` object](TODO: Add link to header section) and specify which variable labels our experimental groups using the `design` argument.
+In this chunk of code, we will specify the `refinebio-treatment` variable to label our experimental groups because it is included in the experimental design as noted in the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/).[@AML]
```

Comment:
```suggestion
In this chunk of code, we will specify the `refinebio_treatment` variable to label our experimental groups because it is included in the experimental design as noted in the accompanying [paper](https://pubmed.ncbi.nlm.nih.gov/28193779/).[@AML]
```
What does this actually mean? Can we switch this to a short description of what the information `refinebio_treatment` represents? aka what was done to those samples?